src/lstm/recodebeam.cpp

```diff
@@ -898,15 +898,13 @@ void RecodeBeamSearch::ContinueContext(
   int length = LengthFromBeamsIndex(index);
   bool use_dawgs = IsDawgFromBeamsIndex(index);
   NodeContinuation prev_cont = ContinuationFromBeamsIndex(index);
-  for (int p = length - 1; p >= 0; --p, previous = previous->prev) {
-    while (previous != nullptr &&
-           (previous->duplicate || previous->code == null_char_)) {
+  for (int p = length - 1; p >= 0 && previous != nullptr; --p) {
+    while (previous->duplicate || previous->code == null_char_) {
       previous = previous->prev;
     }
-    if (previous != nullptr) {
-      prefix.Set(p, previous->code);
-      full_code.Set(p, previous->code);
-    }
+    prefix.Set(p, previous->code);
+    full_code.Set(p, previous->code);
+    previous = previous->prev;
```

Comment:
I added this statement here because otherwise the for statement would have required a line break.

Reply:
> I added this statement here because otherwise the for statement would have required a line break.

I think placing `previous = previous->prev` here or within the `for` statement is equivalent.

Reply:
Yes.